### PR TITLE
[eslint] missing in props validation   react/prop-types

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

Solves eslint errors for react/prop-types and useful for removing the off found in this PR: #7137

### CODE REVIEW

**Comment by driusan** [comment]
This doesn't seem to remove the "off"?

**Comment by maltheism** [comment]
Hi @driusan, I had to do a rebase and now it should be there.

**Review by laemtl — DISMISSED** [review]
@maltheism eslint is not passing :)

**Comment by maltheism** [comment]
Hi @laemtl & @driusan, after rebasing I'm seeing 1194 problems and this is kind of like react's version of type checking. I think this PR should be broken into multiple PRs because it will take awhile to correct all of the problems with likely requiring rebasing overtime.

**Comment by maltheism** [comment]
@driusan @laemtl this PR is ready for review. Some of the prop types might be incorrect but I believe that shouldn't break anything.

**Comment by driusan** [comment]
I've rebased this and it looks like it's failing from new things that have crept in since it was sent.. I think they should be pretty easy to fix so I'll try and find time to fix this PR.

File: jsx/DynamicDataTable.js [hunk 1]
```diff
@@ -151,6 +151,10 @@ class DynamicDataTable extends Component {
 DynamicDataTable.propTypes = {
   DataURL: PropTypes.string.isRequired,
   hiddenHeaders: PropTypes.array,
+  Filter: PropTypes.object,
+  getFormattedCell: PropTypes.func,
+  freezeColumn: PropTypes.string,
+  onSort: PropTypes.bool,
```
Comment (laemtl) [inline jsx/DynamicDataTable.js:157]: It's a PropTypes.func

Comment (driusan) [reply jsx/DynamicDataTable.js]: good catch, fixed it.

**Pull request merged**